src/main/java/com/dawn/jat/illuminati/post/service/PostService.java

```diff
@@ -28,44 +27,79 @@
     @Autowired
     private PostSummaryRepository postSummaryRepository;
 
-    public List<PostSummaryEntity> findPostSummary() {
+    @Autowired
+    private ModelMapper modelMapper;
+
+    public List<PostSummaryEntity> getPostSummaries() {
         return postSummaryRepository.findAll();
     }
 
-    public Optional<PostEntity> findBySlug(String slug) {
-        return postRepository.findBySlug(slug);
+    /**
+     * Get Post by Slug.
+     *
+     * @param slug String
+     * @return PostDto
+     */
+    public PostDto getPostBySlug(String slug) {
+        return modelMapper.map(postRepository.findBySlug(slug), PostDto.class);
     }
 
     /**
      * Create new Entity from postDTO receive from client.
+     *
      * @param postDto PostDto
-     * @return PostEntity
+     * @return PostDto
      */
-    public PostEntity create(PostDto postDto) {
+    public PostDto createPost(PostDto postDto) {
         PostEntity post = new PostEntity();
         post = converter.convertPostDtoToEntity(postDto, post);
-        return postRepository.savePost(post);
+        postRepository.savePost(post);
+        return postDto;
```

Comment:
postDto will not have `id` because we dont have an id when create new Post.

Reply:
Updated